tools/pylib/_boutcore_build/boutcore.pyx.in

```diff
@@ -398,18 +417,22 @@ do
         import numbers
         if isinstance(self,Field3D) and isinstance(other,Field3D):
             fu.cobj=f$n((<Field3D?>self).cobj , (<Field3D?>other).cobj)
+        elif isinstance(self,Field3D) and isinstance(other,Field2D):
+            fu.cobj=f$n((<Field3D?>self).cobj , (<Field2D?>other).cobj)
         elif isinstance(self,Field3D) and isinstance(other, numbers.Number):
             fu.cobj=f$n((<Field3D?>self).cobj , (<double?>float(other)))
         elif isinstance(self,numbers.Number) and isinstance(other, Field3D):
             fu.cobj=f$n(<double?>float(self),(<Field3D?>other).cobj)
         else:
             print("$n",type(self),type(other),isinstance(self,numbers.Number))
-            return NotImplemented
+            raise NotImplemented("Unexpected other - not supported (yet?).")
```

Comment:
Are you ment to raise NotImplementedError?
If so, I guess (some?) of the debug info should be included, to make it possible for a user to understand what types are supported, and what was received ...